### ISSUE TITLE
Option to replace Master Sword at the pedestal

### ISSUE THREAD

**Issue by geekley** [OoTRandomizer/OoT-Randomizer#1477]
You could add an option to **swap Master Sword and Giant's Knife** to make gameplay more interesting.

This would make adult Link be stuck with a breakable weapon until he finds a better sword, similar to how it's possible for child Link to have to stick (ahem) with Deku Sticks until he finds Kokiri Sword. The difference is that a broken knife can still be used as a sword, it's just shorter and deals little damage: 1, same as Kokiri Sword, instead of 4 of biggoron/giant swords, or 2 of Master Sword.

I expect that to cause no problems in logic (unless there's something I'm unaware of that doesn't work with short sword length for adult Link, such as hitting a crystal switch behind bars or something like that).
So here's the suggested behavior for this setting:

**Gameplay changes** with the option enabled
- The Master Sword will be found wherever the Giant's Knife would normally be found; i.e., buy from Medigoron unless he's shuffled, in which case it's randomized in the pool.
- Finding the Master Sword is still required to fight Ganon(dorf). He could reveal it's location, like with Light Arrows.
- The sword at the Pedestal of Time is the Giant's Knife (if Link doesn't have Biggoron Sword).
  The first time Link becomes an adult at the pedestal, he unlocks that instead of the Master Sword.
  If starting age is "adult", Link starts with a broken knife equipped.
- Since Link shouldn't put a broken sword in the pedestal, checking it with a broken sword equipped will instead repair the sword (but you get only 1 hit until it breaks again). After repairing you can check the pedestal again to go back in time.
- Playing Song of Time to the Spiritual Stone altar as an adult (after Door of Time is open) gives you the option to properly restore the health of the Giant's Knife (also unbreaking it if it's broken) for 99 rupees.
  The amount of hits you get depends on how many Spiritual Stones you got in the altar. If none, you get 8 hits, like in vanilla. Emerald multiplies it by 2, Ruby 3 and Sapphire 5. So, when you have all stones, you get 240 hits until it breaks.
    - An additional possibility is to make the price something like 99 - 10 * numberOfMedallions (so it would go down to 39 rupees).

**Visual changes** (if possible)
- The sword model at the pedestal is changed, so it's not (necessarily?) the Master Sword.
- The stones visible at the altar reflect what Link has.

**Advantages**
- This makes the Giant's Knife a lot more interesting in gameplay, since you can repair it easily, you start weak and getting/finding the Master Sword gives you a power upgrade (sort of). In vanilla OoT, the Master Sword makes the knife pretty much pointless, which is kinda disappointing.
- With more hits than just 8, the Giant's Knife can resemble BotW a bit as a "breakable weapon" mechanic.
- Both swords have their pros and cons, and you start with the weakest one, with more gradual upgrades when beating dungeons, even if Master Sword is hard to find.
- It's more challenging, since adult Link starts with half the strength (when broken), not to mention it's harder to hit enemies and you can't use it together with a shield. This also values strategies like to avoid using sword on weaker enemies, to save its hits for the boss fight. Also, finding the Master Sword can be an additional required challenge.

**Comment by ETR-BTF** [comment]
There are three main obstacles to overcome before the Master Sword can be shuffled: glitches that occur when adult Link is swordless need to be fixed, a standalone Master Sword model needs to be made (something that Link can hold over his head), and logic needs to be rewritten (it is currently assumed that adult Link always has a Master Sword). PiMan is working on the logic rewrite, but unfortunately the other two issues are still unaddressed. Master Sword shuffle will likely happen some day, but not anytime soon.

**Comment by geekley** [comment]
> glitches that occur when adult Link is swordless need to be fixed

No, the intention here is to **swap** the swords -- i.e., the sword adult Link starts with and has at all times is the 3rd instead of the 2nd -- so adult Link is never swordless (swordless is cool too, but I assume this is easier to do). Note that a broken knife still cuts! Maybe I was misleading because I forgot to mention: even if starting age is adult, Link still starts with a sword already equipped (the broken knife in this case). I'll edit it to clarify that.

> a standalone Master Sword model needs to be made (something that Link can hold over his head)

Ah I see. I couldn't remember if Ganon fight made the game already have that or not. Is simply scaling it not enough? :thinking:

> logic needs to be rewritten (it is currently assumed that adult Link always has a Master Sword)

Well, with this enabled, it should be assumed that adult Link always has *a* sword (not necessarily the Master Sword). I don't think the game cares about which sword it is anywhere other than the Ganon fight (correct me if I'm wrong). Master Sword would still be required to beat the game (with a logic similar to how Light Arrows are expected).

If swordless adult Link is desired in the future, I think this feature is a good step towards that, with a little bit of compromise to avoid the glitches.

**Comment by DobbyDigital** [comment]
> This would make adult Link be stuck with a breakable weapon until he finds a better sword, similar to how it's possible for child Link to have to stick (ahem) with Deku Sticks until he finds Kokiri Sword. The difference is that a broken knife can still be used as a sword, it's just shorter and deals little damage: 1, same as Kokiri Sword, instead of 4 of biggoron/giant swords, or 2 of Master Sword.
> I expect that to cause no problems in logic (unless there's something I'm unaware of that doesn't work with short sword length for adult Link, such as hitting a crystal switch behind bars or something like that). So here's the suggested behavior for this setting:

I'm not super familiar with damage tables in the game, but the logical differences between Master Sword and Broken Giant's Knife may not be as simple as "everything that Master Sword can do Broken Knife can do as well, just slower".  In the case of Armos, for example, Kokiri Sword (which does the same damage as Broken Knife) does not damage them by itself, whereas Master Sword does. If that were to be the case, logic would need to be adjusted to ensure that adult has some other method for either killing them or stunning them into a state killable with Broken Knife (such as using deku nuts). 



> * Since Link shouldn't put a broken sword in the pedestal, checking it with a broken sword equipped will instead repair the sword (but you get only 1 hit until it breaks again). After repairing you can check the pedestal again to go back in time.
> * Playing Song of Time to the Spiritual Stone altar as an adult (after Door of Time is open) gives you the option to properly restore the health of the Giant's Knife (also unbreaking it if it's broken) for 99 rupees.
>   The amount of hits you get depends on how many Spiritual Stones you got in the altar. If none, you get 8 hits, like in vanilla. Emerald multiplies it by 2, Ruby 3 and Sapphire 5. So, when you have all stones, you get 240 hits until it breaks.
>   
>   * An additional possibility is to make the price something like 99 - 10 * numberOfMedallions (so it would go down to 39 rupees).

All of these changes would require recoding aspects of the game. Likely would have to be in ASM too, since I doubt the Python portions can handle a huge change like this, further limiting the pool of people who would be able to do it. That's not to say it would be impossible, but would be a significant hurdle for implementing.

**Comment by Zannick** [comment]
In any case I'm considering this a subset of Master Sword Shuffle, which I imagine would accomplish the ability to do what you ask (via plando setting the Knife to the pedestal and the Sword to Medigoron).

**Comment by fenhl** [comment]
Closing as a duplicate of the new tracking issue for Master Sword shuffle per the above comment.
